### DISCUSSION TITLE
RC1. Difference between coral and gpu detection

### DISCUSSION THREAD

**Discussion by ediguidedog** [blakeblackshear/frigate#5844] [General]
I've been running various betas on identical hardware to my live setup and while my live box has a USB coral the test box is running with openvino. 

I've consistently notice the detection scores for the betas running openvino is considerably higher than for V11 and the coral and I assumes this was down to a new model. This is good news as it widens the gap between false and real positives.

Today I loaded the RC1 on my live setup with the coral and I expected to see the same higher scores as I had with openvino but they are pretty much the same as V11 using the coral. 

The docs suggest the new model is used by default and adding the model and path entries causes frigate to fail to start so is this difference just down to the different detectors (so 84% on coral seems to be around the same as 99% on openvino) ?

So this works but gives lower scores than openvino 
```
detectors:
    coral:
        type: edgetpu
        device: usb
```

This fails to start 

```
detectors:
    coral:
        type: edgetpu
        device: usb
        model:
            path: "/custom_model.tflite"    
```

**Comment by NickM-27** [comment]
The openvino model is entirely different from the coral model. The coral model has not changed.

Also, scores are entirely down to the model and have nothing to do with the detector.

**Reply by ediguidedog** [reply]
Ah Thanks Nick. I could have sworn I read the new model would be used everywhere.. Are there plans to use the new model with the coral as I find it much better?

**Reply by NickM-27** [reply]
> Ah Thanks Nick. I could have sworn I read the new model would be used everywhere.. Are there plans to use the new model with the coral as I find it much better?

The models are not compatible, they have different arch, different pixel formats, etc.

As far as better models goes, frigate itself is focused on frigate+ for delivering higher quality and more accurate models.

**Reply by blakeblackshear** [reply]
> Are there plans to use the new model with the coral as I find it much better?

The model used by openvino is just the mobilenet ssd model compiled for openvino. Based on my objective measurements against thousands of samples, the current model used by the Coral is better than the mobilenet model for the Coral. Higher scores aren't always better. It does not necessarily mean that the gap will be widened between true and false positives. I'm my experience, the false positives just score higher too. The current model has the widest gap of everything I have tested extensively. 

There may something specific about your cameras that makes one perform better than the other, but don't assume higher scores mean false positives get lower scores.

**Reply by ediguidedog** [reply]
> > Are there plans to use the new model with the coral as I find it much better?
> 
> The model used by openvino is just the mobilenet ssd model compiled for openvino. Based on my objective measurements against thousands of samples, the current model used by the Coral is better than the mobilenet model for the Coral. Higher scores aren't always better. It does not necessarily mean that the gap will be widened between true and false positives. I'm my experience, the false positives just score higher too. The current model has the widest gap of everything I have tested extensively.
> 
> There may something specific about your cameras that makes one perform better than the other, but don't assume higher scores mean false positives get lower scores.

My experience is quite different 
For example on coral and on my drive camera clouds reflected in the windscreen often trigger with a confidence of 75-80%.  People on the otherhand tend to be 75-85%  so there is a significant overlap 

On the openvino reflections are still in a broadly similar range but people are nearly always 95% plus and often 99% so I can pretty much eliminate all false positives.  

Put simply every real person will at some point hit over 85% with openvino but with coral they generally stay firmly in the same percentage bands as clouds and other false positives. In saying that I can manage most of them on a camera by camera basis (with some false positives) but for the beta using openvino I just set some global thresholds and I I don't think I missed a single real event that the V11 setup with coral captured.  For me it was so much better I'm seriously considering not using the coral any more.

**Reply by NickM-27** [reply]
The same model is likely available for the coral on https://github.com/google-coral/test_data/

**Reply by ediguidedog** [reply]
> The same model is likely available for the coral on https://github.com/google-coral/test_data/

Thanks. Any idea which one? I've tried ssdlite_mobiledet_coco_qat_postprocess_edgetpu.tflite as that seemed the closest to the openvino one but I'm not really sure of the naming conventions used and looking at the scores that seems to be the same as the existing one for coral but with slower inference speeds.

**Reply by blakeblackshear** [reply]
I believe it is one of the mobilenet (not mobiledet) models here: https://coral.ai/models/object-detection/

**Reply by ediguidedog** [reply]
> I believe it is one of the mobilenet (not mobiledet) models here: https://coral.ai/models/object-detection/

Found it thanks. I've dropped it in the media folder so it will survive a container recreation. 
My config looks like 

```
# models from https://coral.ai/models/object-detection/    
detectors:
    coral:
        type: edgetpu
        device: usb
        model:
            path: /media/frigate/ssd_mobilenet_v2_coco_quant_postprocess_edgetpu.tflite
model:
    width: 300
    height: 300
    input_tensor: nhwc
    input_pixel_format: bgr
    labelmap_path: /media/frigate/coco_labels.txt  
```
